main/src/ca/uwaterloo/flix/language/phase/optimizer/Inliner.scala

```diff
@@ -371,6 +373,195 @@ object Inliner {
       Expr.NewObject(name, clazz, tpe, eff, methods, loc)
   }
 
+  /**
+    * Evaluate match if possible.
+    *
+    * The match rules are iterated top to bottom, at each step either:
+    *   - Removing the rule if it can never match (either the patten cannot match or the guard is guaranteed false)
+    *   - Removing the match if a rule is known to match (either `Some(12)` matching `Some(x)` or a lenient pattern like `_`)
+    *   - Returning the rules as is if the rule cannot be determined statically.
```

Comment:
What does this mean? Is it more like:

Otherwise leaves the match expression as it were?